### PR TITLE
Update auth example

### PR DESCRIPTION
**Pull request by patsy-issa**
Revamped the auth example ^^

### CODE REVIEW

File: examples/auth/app/controllers/login.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+  @action
+  authenticate(event) {
+    event.preventDefault();
 
-      this.session.authenticate('authenticator:oauth2', this.identification, this.password).catch((reason) => {
-        this.set('errorMessage', reason.error || reason);
+    this.session
+      .authenticate("authenticator:oauth2", this.identification, this.password)
+      .catch((reason) => {
+        this.set("errorMessage", reason.error || reason);
```
Comment (marcoow) [inline examples/auth/app/controllers/login.js:16]: This should not use `set` anymore and `errorMessage` should be a `@tracked` property.

**Comment by marcoow** [comment]
I think you'd also want to run `ember init` to update all of the "infrastructure" files

File: examples/auth/app/controllers/login.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+  @action
+  authenticate(event) {
+    event.preventDefault();
 
-      this.session.authenticate('authenticator:oauth2', this.identification, this.password).catch((reason) => {
-        this.set('errorMessage', reason.error || reason);
+    this.session
+      .authenticate("authenticator:oauth2", this.identification, this.password)
+      .catch((reason) => {
+        this.errorMessage = reason.error || reason;
```
Comment (marcoow) [inline examples/auth/app/controllers/login.js:16]: I think this needs to be `@tracked` in order for the template to update.

Comment (patsy-issa) [reply examples/auth/app/controllers/login.js:18]: added 👍

File: examples/auth/app/controllers/login.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
-import { inject } from '@ember/service';
+import { action } from "@ember/object";
+import { inject } from "@ember/service";
+import Controller from "@ember/controller";
+import { tracked } from "@glimmer/tracking";
 
-export default Controller.extend({
-  session: inject(),
+export default class LoginController extends Controller {
+  @inject()
```
Comment (marcoow) [inline examples/auth/app/controllers/login.js:7]:
```suggestion
  @inject
```

File: examples/auth/app/templates/login.hbs [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+<form {{on "submit" this.authenticate}}>
   <label for="identification">Login</label>
-  <input type="text" id="identification" placeholder="Enter Login" value={{identification}} onchange={{action (mut identification) value="target.value"}}>
+  <Input @id="identification" @type="text" @placeholder="Enter Login" @value={{this.identification}} />
 
   <label for="password">Password</label>
-  <input type="password" id="password" placeholder="Enter Password" value={{password}} onchange={{action (mut password) value="target.value"}}>
+  <Input @id="password" @type="password" @placeholder="Enter Password" @value={{this.password}} />
 
   <button type="submit">Login</button>
```
Comment (marcoow) [inline examples/auth/app/templates/login.hbs:11]: usages of `errorMessage` below have to be changed to `this.errorMessage`

Comment (patsy-issa) [reply examples/auth/app/templates/login.hbs:11]: Already done locally, sorting the last lint error and pushing again, will ping for another review when it is ready 👍

File: examples/auth/app/controllers/application.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
-import Controller from '@ember/controller';
-import { inject } from '@ember/service';
+import { action } from "@ember/object";
+import { inject } from "@ember/service";
+import Controller from "@ember/controller";
 
-export default Controller.extend({
-  session: inject(),
+export default class ApplicationController extends Controller {
+  @inject()
```
Comment (marcoow) [inline examples/auth/app/controllers/application.js:6]:
```suggestion
  @service
```

File: examples/auth/app/controllers/application.js [hunk 6]
```diff
@@ -1,12 +1,13 @@
-import Controller from '@ember/controller';
-import { inject } from '@ember/service';
+import { action } from "@ember/object";
+import { inject } from "@ember/service";
```
Comment (marcoow) [inline examples/auth/app/controllers/application.js:2]:
```suggestion
import { inject as service } from "@ember/service";
```

**Review by marcoow — APPROVED** [review]

**Pull request merged**